pandas/tests/arrays/test_datetimes.py

```diff
@@ -639,3 +641,40 @@ def test_tz_localize_t2d(self):
 
         roundtrip = expected.tz_localize("US/Pacific")
         tm.assert_datetime_array_equal(roundtrip, dta)
+
+    @pytest.mark.parametrize(
+        "error",
+        ["coerce", "raise"],
+    )
+    def test_coerce_fallback(self, error):
+        # GH#46071
+        s = pd.Series(["6/30/2025", "1 27 2024"])
+        expected = pd.Series(
+            [pd.Timestamp("2025-06-30 00:00:00"), pd.Timestamp("2024-01-27 00:00:00")]
+        )
+
+        result = pd.to_datetime(s, errors=error, infer_datetime_format=True)
+
+        if error == "coerce":
+            assert result[1] is not pd.NaT
+
+        tm.assert_series_equal(expected, result)
+
+        expected2 = pd.Series([pd.Timestamp("2000-01-01 00:00:00"), pd.NaT])
+
+        es1 = pd.Series(["1/1/2000", "7/12/1200"])
+        es2 = pd.Series(["1/1/2000", "Hello"])
+
+        if error == "coerce":
```

Comment:
Thanks for the fixes @srotondo. This test is a bit complex and not very easy to read. I think we could make things more readable if we have if for example split this in different tests. Something like:
- `test_to_datetime_infer_datetime_format_with_different_formats`
- `test_to_datetime_infer_datetime_format_with_nat`
- `test_to_datetime_infer_datetime_format_with_out_of_bounds`
- `test_to_datetime_infer_datetime_format_with_invalid_date`

What do you think? Then each test would be quite easy to read. Or maybe by parametrizing the input and the two outputs (coerce and raise). This will avoid a bit of unrepeated code and be quite concise, but probably not so readable.

See what makes sense to you, but would be good to make things as simple as possible, as the pandas codebase is already huge and too complex.

Reply:
@datapythonista thank you for your feedback on the tests.  Because the cases are all closely related, I think it will be easier to understand as one test, but I did simplify some of the test and added more comments to clarify parts of the test.